utils/utils.go

```diff
@@ -413,20 +449,38 @@ func PostUploadProcessFile(serverFileName string, fileName string, tmpFile *os.F
 		log.Println("Error extractiong thumbnail for video " + fileName)
 	} else {
 		log.Println("Extracted thumbnail " + extractedThumbnail)
-		uploadThumbnailToAppwrite(fileName, db)
+		err = uploadThumbnailToAppwrite(fileName, db)
+		if err != nil {
+			log.Printf("Error uploading thumbnail to Appwrite Storage for video %s : %v", fileName, err)
+		}
 	}
 
 	log.Println("Breaking the video into .ts files.")
 
-	breakResult := breakFile(("./video/" + serverFileName), fileName)
+	breakResult, err := breakFile(("./video/" + serverFileName), fileName)
 	if breakResult {
-		log.Println("Successfully broken " + fileName + " into .ts files.")
+		log.Println("Successfully broken " + fileName + "into .ts files.")
 		log.Println("Deleting the original file from server.")
-		closeVideoFile(tmpFile)
-		uploadToAppwrite(fileName, db)
-		log.Println("Successfully uploaded chunks of", fileName, "to Appwrite Storage")
+		err = closeVideoFile(tmpFile)
+		if err != nil {
+			log.Printf("Error closing temporary file %s: %v", tmpFile.Name(), err)
+			if err := updateUploadStatus(db, fileName, types.UploadFailed); err != nil {
+				log.Printf("Error updating upload status for video %s in DB: %v", fileName, err)
+			}
+		}
+		err = uploadToAppwrite(fileName, db)
+		if err != nil {
+			log.Printf("Error uploading chunks of %s to Appwrite Storage : %v", fileName, err)
+			if err := updateUploadStatus(db, fileName, types.UploadFailed); err != nil {
+				log.Printf("Error updating upload status for video %s in DB: %v", fileName, err)
+			}
+		}
+		log.Printf("Successfully uploaded chunks of %s to Appwrite Storage", fileName)
```

Comment:
This log also comes even if the upload has failed.
This should be inside an else block.
Rest changes look good to me.